rotkehlchen/api/v1/fields.py

```diff
@@ -380,22 +392,38 @@ def _deserialize(
     ) -> Asset:
         if isinstance(value, str) is False:
             raise ValidationError(f'Tried to initialize an asset out of a non-string identifier {value}')  # noqa: E501
-
+        # Since the identifier could be url encoded for evm tokens in urls we need to unquote it
+        real_value: str = urllib.parse.unquote(value)
         try:
-            asset = Asset(
-                identifier=urllib.parse.unquote(value),
-                form_with_incomplete_data=self.form_with_incomplete_data,
-            )
-        except (DeserializationError, UnknownAsset) as e:
+            if self.expected_type == Asset:
+                if real_value.startswith(NFT_DIRECTIVE):
+                    asset = Asset(identifier=real_value)
+                else:
+                    # Just to check identifier's existence
+                    asset = Asset(identifier=real_value).resolve_to_asset_with_name_and_type()
+            elif self.expected_type == AssetWithOracles:
+                asset = Asset(identifier=real_value).resolve_to_asset_with_oracles()
+            elif self.expected_type == CryptoAsset:
+                asset = CryptoAsset(real_value)
```

Comment:
CryptoAsset and EvmToken when initialized will resolve to the corresponding asset. So `CryptoAsset(real_value)` will be a final asset and the same with `EvmToken`. The reason is needed above is because they are being initialized from `Asset`